expect.go

```diff
@@ -78,16 +87,31 @@ func (c *Console) Expect(opts ...ExpectOpt) (string, error) {
 	}()
 
 	for {
+		var readCh chan struct{}
 		if readTimeout != nil {
-			err = c.passthroughPipe.SetReadDeadline(time.Now().Add(*readTimeout))
-			if err != nil {
-				return buf.String(), err
-			}
+			readCh = make(chan struct{})
+			go func() {
+				select {
+				case <-readCh:
+				case <-time.After(*readTimeout):
+					c.cancelReader.Cancel()
+				}
```

Comment:
Let's use a timer we can explicitly stop here:

```suggestion
				timer := time.NewTimer(*readTimeout)
				select {
				case <-readCh:
					timer.Stop()
				case <-timer.C:
					c.cancelReader.Cancel()
				}
```

Otherwise, the timers created by `time.After` will stay active after a read finishes. If reads are happening very fast, this can cause a lot of unnecessary timers to accumulate.